scripts/west_commands/sign.py

```diff
@@ -633,3 +648,80 @@ def sign(self, command, build_dir, build_conf, formats):
+        return commandertool
+
+    def get_security_configs(self, build_conf, args, command):
+        # Retrieve configurations, prioritizing command-line args, then build_conf
+        siwx91x_ota_key = getattr(args, 'siwx91x-ota-key', build_conf.get('CONFIG_SIWX91X_OTA_KEY'))
+        siwx91x_private_key = getattr(args, 'siwx91x-private-key', build_conf.get('CONFIG_SIWX91X_PRIVATE_KEY'))
+
+        # Refer find_commandertool() function to identify the Commander tool.
+        commander_path = self.find_commandertool(command, args)
+        # Validate required settings
+        if not siwx91x_ota_key:
+            command.die("SIWX91X_OTA_KEY not provided. Use --siwx91x-ota-key=your_key or add CONFIG_SIWX91X_OTA_KEY in prj.conf.")
+        if not siwx91x_private_key:
+            command.die("SIWX91X_PRIVATE_KEY not provided. Use --siwx91x-private-key=/path/to/key or add CONFIG_SIWX91X_PRIVATE_KEY in prj.conf.")
+
+        # Validate private key path
+        if not os.path.isfile(siwx91x_private_key):
+            command.die(f"Private key not found at {siwx91x_private_key}. Please ensure the path is correct.")
+
+        return siwx91x_ota_key, commander_path, siwx91x_private_key
+
+    def sign(self, command, build_dir, build_conf, formats):
+        """Sign the Zephyr binary using Silicon Labs Commander.
+
+        :param command: The Sign instance (provides args and utility methods)
+        :param build_dir: The build directory path
+        :param build_conf: BuildConfiguration object for the build directory
+        :param formats: List of formats to generate (e.g., ['bin', 'rps'])
+        """
+        self.command = command
+        args = command.args
+        b = pathlib.Path(build_dir)
+
+        # Check if signing is needed
+        if not formats:
+            if not args.quiet:
+                command.inf(f"No output formats specified, not signing {args}")
+            return
+
+        # Setting the input and output paths
+        input_rps = b / 'zephyr' / "zephyr.bin.rps"
+        output_rps = args.sbin or (b / 'zephyr' / "zephyr_signed.bin.rps")
+
+        # Check if input binary exists
+        if not input_rps.is_file():
+            command.die(f"No .rps found at {input_rps}. Ensure the build generated zephyr.bin.rps in {b / 'zephyr'}")
+
+        # Load configuration
+        siwx91x_ota_key, commander_path, siwx91x_private_key = self.get_security_configs(build_conf, args, command)
+
+        # Build the Silicon Labs Commander signing command
+        sign_base = [
+            commander_path,
+            "rps",
+            "convert",
+            str(output_rps),
+            "--app", str(input_rps),
+            "--mic", siwx91x_ota_key,
+            "--encrypt", siwx91x_ota_key,
+            "--sign", siwx91x_private_key,
+            *args.tool_args
+        ]
+
+        print("Running command:", ' '.join(sign_base))
```

Comment:
Don't use `print`, use regular logging commands like `command.inf()` like everywhere else in this file.